# QUESTION 6

For the passengers picked up in the zone named "East Harlem North" in November 2025, which was the drop off zone that had the largest tip? (1 point) <br>

a. JFK Airport <br>
b. Yorkville West <br>
c. East Harlem North <br>
d. LaGuardia Airport

In [11]:
#download dataset green trip data
!wget https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2025-11.parquet

--2026-01-26 06:30:03--  https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2025-11.parquet
Resolving d37ci6vzurychx.cloudfront.net (d37ci6vzurychx.cloudfront.net)... 52.85.97.13, 52.85.97.44, 52.85.97.61, ...
Connecting to d37ci6vzurychx.cloudfront.net (d37ci6vzurychx.cloudfront.net)|52.85.97.13|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1164775 (1.1M) [binary/octet-stream]
Saving to: ‘green_tripdata_2025-11.parquet.1’

green_tripdata_2025 100%[===================>]   1.11M  --.-KB/s    in 0.03s   

2026-01-26 06:30:03 (31.9 MB/s) - ‘green_tripdata_2025-11.parquet.1’ saved [1164775/1164775]



In [12]:
#download dataset zone data
!wget https://github.com/DataTalksClub/nyc-tlc-data/releases/download/misc/taxi_zone_lookup.csv

--2026-01-26 06:30:03--  https://github.com/DataTalksClub/nyc-tlc-data/releases/download/misc/taxi_zone_lookup.csv
Resolving github.com (github.com)... 140.82.116.4
Connecting to github.com (github.com)|140.82.116.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://release-assets.githubusercontent.com/github-production-release-asset/513814948/5a2cc2f5-b4cd-4584-9c62-a6ea97ed0e6a?sp=r&sv=2018-11-09&sr=b&spr=https&se=2026-01-26T07%3A18%3A24Z&rscd=attachment%3B+filename%3Dtaxi_zone_lookup.csv&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2026-01-26T06%3A17%3A36Z&ske=2026-01-26T07%3A18%3A24Z&sks=b&skv=2018-11-09&sig=eNC1Hxn%2FaHFFo7adgaN6Nhydwa%2BncmkfK%2By%2B1aEhp8U%3D&jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc2OTQwOTMwMywibmJmIjoxNzY5NDA5MDAzLCJwYXRoIjoicmVsZWFzZWF

In [13]:
import pandas as pd
trip_data = pd.read_parquet("green_tripdata_2025-11.parquet")
zone_data = pd.read_csv("taxi_zone_lookup.csv")

In [14]:
#checking the data sample
trip_data.head(2)

VendorID lpep_pickup_datetime lpep_dropoff_datetime store_and_fwd_flag  \
0         2  2025-11-01 00:34:48   2025-11-01 00:41:39                  N   
1         2  2025-11-01 00:18:52   2025-11-01 00:24:27                  N   

   RatecodeID  PULocationID  DOLocationID  passenger_count  trip_distance  \
0         1.0            74            42              1.0           0.74   
1         1.0            74            42              2.0           0.95   

   fare_amount  ...  mta_tax  tip_amount  tolls_amount  ehail_fee  \
0          7.2  ...      0.5        1.94           0.0        NaN   
1          7.2  ...      0.5        0.00           0.0        NaN   

   improvement_surcharge  total_amount  payment_type  trip_type  \
0                    1.0         11.64           1.0        1.0   
1                    1.0          9.70           2.0        1.0   

   congestion_surcharge  cbd_congestion_fee  
0                   0.0                 0.0  
1                   0.0                 0.0  

[2 rows x 21 columns]

In [15]:
#checking the data sample
zone_data.head(2)

LocationID Borough            Zone service_zone
0           1     EWR  Newark Airport          EWR
1           2  Queens     Jamaica Bay    Boro Zone

In [16]:
#Identify the Dropout Zone by merging zone_data with trip data
df_dropout_zone = pd.merge(trip_data,
    zone_data,
    left_on="DOLocationID",
    right_on="LocationID",
    how="left"
)

df_dropout_zone.head(3)

VendorID lpep_pickup_datetime lpep_dropoff_datetime store_and_fwd_flag  \
0         2  2025-11-01 00:34:48   2025-11-01 00:41:39                  N   
1         2  2025-11-01 00:18:52   2025-11-01 00:24:27                  N   
2         2  2025-11-01 01:03:14   2025-11-01 01:15:24                  N   

   RatecodeID  PULocationID  DOLocationID  passenger_count  trip_distance  \
0         1.0            74            42              1.0           0.74   
1         1.0            74            42              2.0           0.95   
2         1.0            83           160              1.0           2.19   

   fare_amount  ...  improvement_surcharge  total_amount  payment_type  \
0          7.2  ...                    1.0         11.64           1.0   
1          7.2  ...                    1.0          9.70           2.0   
2         13.5  ...                    1.0         21.00           1.0   

   trip_type  congestion_surcharge  cbd_congestion_fee  LocationID    Borough  \
0        1.0                   0.0                 0.0          42  Manhattan   
1        1.0                   0.0                 0.0          42  Manhattan   
2        1.0                   0.0                 0.0         160     Queens   

                   Zone  service_zone  
0  Central Harlem North     Boro Zone  
1  Central Harlem North     Boro Zone  
2        Middle Village     Boro Zone  

[3 rows x 25 columns]

In [17]:
#Identify the Pickup Zone by merging zone_data with df_dropout_zone
df_pickup_zone = pd.merge(df_dropout_zone,
    zone_data,
    left_on="PULocationID",
    right_on="LocationID",
    how="left"
)

df_pickup_zone.head(3)

VendorID lpep_pickup_datetime lpep_dropoff_datetime store_and_fwd_flag  \
0         2  2025-11-01 00:34:48   2025-11-01 00:41:39                  N   
1         2  2025-11-01 00:18:52   2025-11-01 00:24:27                  N   
2         2  2025-11-01 01:03:14   2025-11-01 01:15:24                  N   

   RatecodeID  PULocationID  DOLocationID  passenger_count  trip_distance  \
0         1.0            74            42              1.0           0.74   
1         1.0            74            42              2.0           0.95   
2         1.0            83           160              1.0           2.19   

   fare_amount  ...  congestion_surcharge  cbd_congestion_fee  LocationID_x  \
0          7.2  ...                   0.0                 0.0            42   
1          7.2  ...                   0.0                 0.0            42   
2         13.5  ...                   0.0                 0.0           160   

   Borough_x                Zone_x  service_zone_x  LocationID_y  Borough_y  \
0  Manhattan  Central Harlem North       Boro Zone            74  Manhattan   
1  Manhattan  Central Harlem North       Boro Zone            74  Manhattan   
2     Queens        Middle Village       Boro Zone            83     Queens   

              Zone_y  service_zone_y  
0  East Harlem North       Boro Zone  
1  East Harlem North       Boro Zone  
2   Elmhurst/Maspeth       Boro Zone  

[3 rows x 29 columns]

In [18]:
#rename the column
df_2 = df_pickup_zone.rename(columns={"Zone_x": "zone_drop", "Zone_y": "zone_pickup"})

In [19]:
#use groupby to calculate tip amount per zone drop
df_2.groupby("zone_drop").agg(total=("tip_amount", "sum")).sort_values("total", ascending=False)

total
zone_drop                         
LaGuardia Airport          7254.51
Upper East Side North      6964.59
Upper West Side North      5472.67
East Harlem South          4476.26
Upper West Side South      4109.34
...                            ...
Madison                       0.00
Starrett City                 0.00
South Beach/Dongan Hills      0.00
Saint George/New Brighton     0.00
Van Cortlandt Park            0.00

[247 rows x 1 columns]

The drop off zone that had the largest tip is <b> La Guardia Airport (d) </b>, which has value of 7254.51